geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ClientUserAuths.java

```diff
@@ -49,10 +50,18 @@ public Long putUserAuth(UserAuthAttributes userAuthAttr) {
     return newId;
   }
 
-  public Long putSubject(Subject subject) {
-    final Long newId = getNextID();
+  public Long putSubject(Subject subject, long existingUniqueId) {
+    final Long newId;
+    if (existingUniqueId == 0 || existingUniqueId == -1) {
+      newId = getNextID();
+    } else {
+      // the user re-authenticate back, remove the existing subject, reuse the uniqueId
+      removeSubject(existingUniqueId);
+      newId = existingUniqueId;
+    }
+
     uniqueIdVsSubject.put(newId, subject);
-    logger.debug("Subject of {} added.", newId);
+    logger.info("Subject of {} added.", newId);
```

Comment:
Do we want this to always log? It was debug before.

Reply:
Oh, no, accidental change.